homework/src/index.js

```diff
@@ -31,14 +34,83 @@
+        const repositoryDiv = createAndAppend('div', root, { class: 'repositoryDiv' });
+        const header = createAndAppend('header', repositoryDiv, { class: 'topping' });
+        createAndAppend('h1', header, { html: 'HYF Repositories ', class: 'hyfRepo' });
+        const selectBox = createAndAppend('select', header, { class: 'select-box' });
+        data.sort(function (a, b) { return a.name.localeCompare(b.name); });
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selectBox, { value: i, html: data[i].name });
+        }
+        const infoBox = createAndAppend('div', root, { class: 'infoBox' });
+        const ContributorBox = createAndAppend('div', root, { class: 'ContributorBox' });
+        selectBox.addEventListener('change', () => {
+          const value = data[selectBox.selectedIndex].url;
+          fetchJSON(value)
+            .then(data => {
+              repositoryInfo(infoBox, data, value);
+            });
+          const contributors = data[selectBox.selectedIndex].contributors_url;
+          fetchJSON(contributors)
+            .then(data => {
+              ContributorsInfo(ContributorBox, data, contributors);
+            });
+        });
+        fetchJSON(data[0].url)
+          .then(data => {
+            repositoryInfo(infoBox, data);
+          });
+        fetchJSON(data[0].contributors_url)
+          .then(data => {
+            ContributorsInfo(ContributorBox, data);
+          });
+      })
+      .catch(err => {
+        const root = document.getElementById('root');
+        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+
+      });
+  }
+
+  //adding rows function
+  function addRow(label, value, tbody) {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { html: label });
+    createAndAppend('td', row, { html: value });
+  }
+
+  //information box of each repository
+  function repositoryInfo(infoBox, data) {
+    infoBox.innerHTML = '';
+    const table = createAndAppend('table', infoBox, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    addRow('Repository', data.name, tbody);
+
+    if (data.description) {
+      addRow('Description', data.description, tbody);
+    }
+
+    addRow('Forks', data.forks, tbody);
+    addRow('Updated', data.updated_at, tbody);
+  }
+
+  function ContributorsInfo(ContributorBox, contributors) {
+    ContributorBox.innerHTML = '';
+    const ContributorTable = createAndAppend('div', ContributorBox, { class: 'ContributorTable' });
+    createAndAppend('p', ContributorTable, { html: 'Contributors', class: 'contTitle' });
+    const ul = createAndAppend('ul', ContributorTable, { class: 'contributorsList' });
+    if (contributors !== null) {
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', ul, { href: (`${contributor.html_url}`), target: '_blank' });
```

Comment:
This is a comment I made in week 1 which is still not corrected:

A `<ul>` element can only have `<li>` element as its children. If cannot have an `<a>` or any tag other than `<li>`.  But your code generates this structure:

```html
<ul class="contributorsList">
  <a href="https://github.com/mkruijt" target="_blank">
    <li class="listItem"><img src="https://avatars2.githubusercontent.com/u/7113309?v=4" class="pictures">
      <div class="contributorName">mkruijt</div>
      <div class="badgeNr">246</div>
    </li>
  </a>
  ...
</ul>
```